pkg/addon-operator/operator.go

```diff
@@ -1256,6 +1256,16 @@ func (op *AddonOperator) HandleModuleHookRun(t sh_task.Task, labels map[string]s
 	hm := task.HookMetadataAccessor(t)
 	taskHook := op.ModuleManager.GetModuleHook(hm.HookName)
 
+	err := taskHook.GetRateLimiter().Wait(context.Background())
```

Comment:
Maybe we can hide the getter method `GetRateLimiter()` to forbid other interactions with the rate limiter.

Before:
```go
func (t *taskHook) GetRateLimiter() *rate.Limiter {
  return t.RateLimiter
}
```

After:
```go
func (t *taskHook) RateLimitWait() err {
  return t.rateLimiter.Wait(context.Background())
}
```

Reply:
The context should be exposed, so it's more like that:

```go
func (t *taskHook) RateLimitWait(ctx context) err {
  return t.rateLimiter.Wait(ctx)
}
```

`RateLimiter` is public to use it from shell-operator and from addon-operator.